test/framework/daemonset_helpers.go

```diff
@@ -60,3 +62,39 @@ func WaitForKubeProxyUpgrade(ctx context.Context, input WaitForKubeProxyUpgradeI
 		return false, nil
 	}, intervals...).Should(BeTrue())
 }
+
+// WatchDaemonSetLogsByLabelSelectorInput is the input for WatchDaemonSetLogsByLabelSelector.
+type WatchDaemonSetLogsByLabelSelectorInput struct {
+	GetLister GetLister
+	Cache     toolscache.Cache
+	ClientSet *kubernetes.Clientset
+	Labels    map[string]string
+	LogPath   string
+}
+
+// WatchDaemonSetLogsByLabelSelector streams logs for all containers for all pods belonging to a daemonset on the basis of label. Each container's logs are streamed
+// in a separate goroutine so they can all be streamed concurrently. This only causes a test failure if there are errors
+// retrieving the daemonset, its pods, or setting up a log file. If there is an error with the log streaming itself,
+// that does not cause the test to fail.
+func WatchDaemonSetLogsByLabelSelector(ctx context.Context, input WatchDaemonSetLogsByLabelSelectorInput) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for WatchDaemonSetLogsByLabelSelector")
+	Expect(input.Cache).NotTo(BeNil(), "input.Cache is required for WatchDaemonSetLogsByLabelSelector")
+	Expect(input.ClientSet).NotTo(BeNil(), "input.ClientSet is required for WatchDaemonSetLogsByLabelSelector")
+	Expect(input.Labels).NotTo(BeNil(), "input.Selector is required for WatchDaemonSetLogsByLabelSelector")
+
+	daemonSetList := &appsv1.DaemonSetList{}
+	Eventually(func() error {
+		return input.GetLister.List(ctx, daemonSetList, client.MatchingLabels(input.Labels))
+	}, retryableOperationTimeout, retryableOperationInterval).Should(Succeed(), "Failed to get daemonsets for labels")
+
+	for _, daemonSet := range daemonSetList.Items {
+		watchPodLogs(ctx, watchPodLogsInput{
+			Cache:          input.Cache,
```

Comment:
Of these parameters only Cache and LabelSelector are actually used. Let's drop all others from watchPodLogsInput

(this way we can also avoid setting a DS name as DepoymentName)

Reply:
Please also drop the corresponding fields of input structs of funcs calling watchPodLogs

Reply:
Let's do the changes to watchPodLogs and WatchDaemonSetLogsByLabelSelector in this PR (so we can cherry-pick)

and then the changes to input structs of other callers in a separate PR

Reply:
oh well they are all used below inside the `streamPodLogs` functions. But great point about the `DeploymentName` part, I will try to get a better name for it.

Reply:
Oh ups, my bad. 

Not a super great pattern to pass through input structs meant for one func to another one :D

Reply:
But doesn't matter much, all internal funcs